### PR TITLE
add fusepp by jachappell/Fusepp

### PR DESCRIPTION
**Pull request by winternet**
Specify library name and version:  **fusepp/3.0.0**

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — COMMENTED** [review]
👋 Thanks for sharing this really cool library!

We typically do not add forks as they are harder for us to maintain.

In this particular case I think we can avoid it

- For the version checkout https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases
- For the CMake checkout: https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#no-upstream-build-scripts 
   - [here's an example](https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/http_parser/all/CMakeLists.txt)

If you need more help, feel free to @ me 😄

**Comment by winternet** [comment]
Hi, makes sense - thanks for your feedback! I adapted the package accordingly.

> wave Thanks for sharing this really cool library!
> 
> We typically do not add forks as they are harder for us to maintain.
> 
> In this particular case I think we can avoid it
> 
> * For the version checkout https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases
> * For the CMake checkout: https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#no-upstream-build-scripts
>   
>   * [here's an example](https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/http_parser/all/CMakeLists.txt)
> 
> If you need more help, feel free to @ me smile

File: recipes/fusepp/all/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.4)
+project(cmake_wrapper)
+
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup()
```
Comment (uilianries) [inline recipes/fusepp/all/CMakeLists.txt:5]:
```suggestion
conan_basic_setup(KEEP_RPATHS)
```

File: recipes/fusepp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("Fuse*.h", dst="include", src=self._source_subfolder)
+        self.copy("Fuse.cpp", dst="include", src=self._source_subfolder)
+        self.copy("*.dll", dst="bin", keep_path=False)
+        self.copy("*.so", dst="lib", keep_path=False)
+        self.copy("*.dylib", dst="lib", keep_path=False)
+        self.copy("*.a", dst="lib", keep_path=False)
```
Comment (uilianries) [inline recipes/fusepp/all/conanfile.py:66]:
```suggestion
        cmake = self._configure_cmake()
        cmake.install()
```

If you are providing a cmake file, use install feature.

File: recipes/fusepp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def requirements(self):
+        self.requires("libfuse/3.10.5")
+
```
Comment (uilianries) [inline recipes/fusepp/all/conanfile.py:53]:
```suggestion

    def _configure_cmake(self):
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.configure()
        return self._cmake

```

**Comment by uilianries** [comment]
Simplify the test package to a single and very simple cpp file. The idea is testing library linkage, not run an example or unit tests. Take a look on others recipes in Conan Center Index.

**Review by SSE4 — DISMISSED** [review]

File: recipes/fusepp/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
```
Comment (uilianries) [inline recipes/fusepp/all/conanfile.py:69]:
```suggestion
        cmake.install()

    def package_info(self):
        self.cpp_info.libs = ["fusecpp"]
        # TODO: Remove after Conan 2.0
        self.names["cmake_find_package"] = "fusecpp"
        self.names["cmake_find_package_multi"] = "fusecpp"

         self.set_property("cmake_file_name", "fusecpp")
         self.set_property("cmake_target_name", "fusecpp")
```

Need to export generated libraries

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**